homeassistant/components/evohome/__init__.py

```diff
@@ -352,8 +502,26 @@ def __init__(self, evo_broker, evo_device) -> None:
         self._device_state_attrs = {}
 
     @callback
-    def _refresh(self) -> None:
-        self.async_schedule_update_ha_state(force_refresh=True)
+    def _refresh(self, payload=None) -> None:
+        if payload is None:
+            self.async_schedule_update_ha_state(force_refresh=True)
+        elif payload["unique_id"] == self._unique_id:
+            if payload["service"] in [SVC_SET_ZONE_OVERRIDE, SVC_RESET_ZONE_OVERRIDE]:
+                self.hass.async_create_task(
```

Comment:
We should make `_refresh` an async function and have it call these functions directly. It will give us better stacktraces.